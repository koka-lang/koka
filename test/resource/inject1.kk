// Test inject

fun foo(action : () -> e a ) : <ndet|e> a
{
  val x = inject<ndet>(action)
  if (random-bool()) then foo(action) else x 
}

fun main() {
  foo{42}.println
}