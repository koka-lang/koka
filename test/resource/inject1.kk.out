42
foo : forall<a,e> (action : () -> <div|e> a) -> <div,ndet|e> a
main : () -> <div,console,ndet> ()